packages/trpc/server/routers/viewer/organizations/adminDelete.handler.ts

```diff
@@ -0,0 +1,80 @@
+import { deleteDomain } from "@calcom/lib/domainManager/organization";
+import logger from "@calcom/lib/logger";
+import { prisma } from "@calcom/prisma";
+
+import { TRPCError } from "@trpc/server";
+
+import type { TrpcSessionUser } from "../../../trpc";
+import type { TAdminDeleteInput } from "./adminDelete.schema";
+
+const log = logger.getSubLogger({ prefix: ["organizations/adminDelete"] });
+type AdminDeleteOption = {
+  ctx: {
+    user: NonNullable<TrpcSessionUser>;
+  };
+  input: TAdminDeleteInput;
+};
+
+export const adminDeleteHandler = async ({ input }: AdminDeleteOption) => {
+  const foundOrg = await prisma.team.findUnique({
+    where: {
+      id: input.orgId,
+      metadata: {
+        path: ["isOrganization"],
+        equals: true,
+      },
+    },
+    include: {
+      members: {
+        select: {
+          user: true,
+        },
+      },
+    },
+  });
+
+  if (!foundOrg)
+    throw new TRPCError({
+      code: "FORBIDDEN",
+      message: "Organization not found",
+    });
+
+  if (foundOrg.slug) {
+    await deleteDomain(foundOrg.slug);
+  }
+
+  await renameUsersToAvoidUsernameConflicts(foundOrg.members.map((member) => member.user));
+  await prisma.team.delete({
+    where: {
+      id: input.orgId,
+    },
+  });
+
+  return {
+    ok: true,
+    message: `Organization ${foundOrg.name} deleted.`,
+  };
+};
+
+export default adminDeleteHandler;
+
+async function renameUsersToAvoidUsernameConflicts(users: { id: number; username: string | null }[]) {
```

Comment:
no need to have any custom check for premium username here I suppose ?

Reply:
Yeah I don't think so. I don't think premium usernames are anyways applicable for an organization.